doc/docportal/advanced_topics/understand_search_box.rst

```diff
@@ -0,0 +1,137 @@
+
+For example, if you type ``m`` you would get all the games containing letter 'M' in description.
+
+If you type ``monkey``, you would get games like "The Curse of Monkey Island (Demo/Windows)", "Infinite Monkeys", "Three Monkeys, One Cage", etc.
+
+This is what you need in most simple cases, but is not enough to search for more complex things.
+
+Let's search the configuration key's values
+___________________________________________
+
+Game properties listed at :doc:`configuration_file` can be checked with the filter as well.
+
+Advanced patterns
+*******************************************
+
+There are currently 3 types of patterns, that can be used for accessing configuration file:
+
+.. csv-table::
+  	:header-rows: 1
+	:class: config
+
+		Type,Description,Example,Example result
+		"<key>=<value>", check exact value of ``<key>``, ``gameid=reversion2``, games with exact gameid "reversion2"
+		"<key>:<value>", search substring at ``<key>``, ``description::``, games with ":" substring at description
+		"<key>~<value>", match wildcard against ``<key>``, ``path~D:\\games\\*``, games located at Windows folder D:\\Games\\
+
+Available configuration keys
+*******************************************
+
+.. note::
+
+	The ``\`` character used at Windows paths should be escaped as ``\\`` when using wildcards.
+
+You can use any :ref:`configuration_keys` as a ``<key>`` part of the search pattern.
+
+Here are some more examples:
+
+- ``show_fps=true`` - games with "Show FPS" option set to true
+- ``extra~?*`` - games with non-empty extra part of the description
+- ``guioptions:noLang`` - games without displayed language
+- ``keymap_engine-default_LCLK:MOUSE_RIGHT`` - games with right mouse button remapped to left mouse button
+
+Abbrivating configuration keys
+*******************************************
+
+Some common used configuration keys can be abbrivated, just type any prefix instead of full strings for those 6 keys:
+
+.. csv-table::
+  	:header-rows: 1
+	:class: config
+
+		Full key,Description,Examples
+		``description``, game description as displayed at the game list,"
+
+	- ``d:monkey``
+	- ``desc:monkey``
+	- ``descr:monkey``
+	- ``description:monkey``"
+		``engineid``, internal ID of the game engine,"
+
+	- ``e:ags``
+	- ``engine:monkey``
+	- ``engineed:monkey``"
+		``gameid``, internal ID of the game,"
+
+	- ``g:monkey``
+	- ``game:monkey``
+	- ``gameid:monkey``"
+		``language``, internal ID of the language (usually 2 letter code like "de"/"en"/"fr"/"jp"/etc),"
+
+	- ``l=en``
+	- ``lang=en``
+	- ``language=en``"
+		``path``, Filesystem path for the game,"
+
+	- ``p~D:*``
+	- ``path~D:*``"
+		``path``, Filesystem path for the game,"
+
+	- ``pl~windows``
+	- ``platform~windows``"
+
+.. note::
+
+	The ``platform`` key can't be abbrivated to ``p``, since ``p`` is already used for ``path``.
```

Comment:
Interesting idea, but that conflicts the point of having all configuration keys mapped to aptual keys at scummvm.ini file.
So, `path` would be supported anyway and it may be strange abbrivating `path` to `f`.

Also, from the list "f for file, filepath, filesystem, or filesystem-path" I suspect that my idea of abbrivating was actually lost inside translation. Current code allows you to type ANY number of characters of abbrivated configuration key. For `language` I suggest `l`, `lang` and `language` at Examples column, but you can also use `la:en` or `langu:en` or `languag:en`. Keys `filepath` and `filesystem` don't follow this pattern. Should I emphasis this somehow?

Reply:
Interesting idea, but that conflicts the point of having all configuration keys mapped to aptual keys at scummvm.ini file.
So, `path` would be supported anyway (since there is such key in scummvm.ini) and it may be strange abbrivating path to `f`.

Also, from the list "f for file, filepath, filesystem, or filesystem-path" I suspect that my idea of abbrivating was actually lost inside translation. Current code allows you to type ANY number of characters of abbrivated configuration key. For `language` I suggest `l`, `lang` and `language` at `Examples` column, but you can also use `la:en` or `langu:en` or `languag:en`, etc. Keys `filepath` and `filesystem` don't follow this pattern - they have different suffixes. Should I emphasis this somehow?